extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -399,30 +405,29 @@ export class AmpConsent extends AMP.BaseElement {
         config['promptIfUnknownForGeoGroup'],
     'neither checkConsentHref nor ' +
     'promptIfUnknownForGeoGroup is defined');
+    let geoPromise = Promise.resolve(null);
     let remoteConfigPromise = Promise.resolve(null);
-    if (config['checkConsentHref']) {
-      remoteConfigPromise = this.getConsentRemote_(instanceId);
-      this.passSharedData_(instanceId, remoteConfigPromise);
-    }
-    let geoPromise = Promise.resolve();
     if (config['promptIfUnknownForGeoGroup']) {
       const geoGroup = config['promptIfUnknownForGeoGroup'];
       geoPromise = this.isConsentRequiredGeo_(geoGroup);
+    } else {
+      remoteConfigPromise = this.getConsentRemote_(instanceId);
     }
+
     return geoPromise.then(promptIfUnknown => {
       return remoteConfigPromise.then(response => {
```

Comment:
Couldn't we check here for promptIfUnkown?

If we have the prompt, we can resolve, and not wait for the remote config.

And then `this.isPromptRequired`, could become `this.isPromptRequiredForRemoteConifg()`. Or inlined in the `.then`

Reply:
Good point. `isPromptRequired` It was there because we need to wait for both promise to resolve to determine. Since we can resolve on the geoPromise along right now, it's going to be cleaner : ) 
done